app/utils/testing.py

```diff
@@ -1,16 +1,37 @@
+from typing import Dict
+
+from django.db.models.base import ModelBase
+from django.http import HttpResponse
+from django.template.context import Context
 from django.urls import reverse
 from django.utils.http import urlencode
 
 
-def reverse_with_query(name, query):
+def reverse_with_query(viewname: str, query: Dict[str, str], *args, **kwargs):
```

Comment:
I think for query you should either use `dict` or `collections.abc.Mapping` if you want a more permissive type. Generally importing `typing` for data structures like this is deprecated. See https://docs.python.org/3/library/typing.html#aliases-to-built-in-types.